tests/e2e/specs/merchant/order-full-refund.spec.js

```diff
@@ -0,0 +1,134 @@
+
+		// Get the order ID so we can open it in the merchant view
+		const orderIdField = await page.$(
+			'.woocommerce-order-overview__order.order > strong'
+		);
+		orderId = await orderIdField.evaluate( ( el ) => el.innerText );
+
+		// Login and open the order
+		await StoreOwnerFlow.login();
+		await StoreOwnerFlow.goToOrder( orderId );
+
+		// We need to remove any listeners on the `dialog` event otherwise we can't catch the dialog below
+		await page.removeAllListeners( 'dialog' );
+
+		// Get the order price
+		const priceElement = await page.$( '.woocommerce-Price-amount' );
+		orderAmount = await page.evaluate(
+			( el ) => el.textContent,
+			priceElement
+		);
+	} );
+
+	afterAll( async () => {
+		await StoreOwnerFlow.logout();
+	} );
+
+	it( 'should process a full refund for an order', async () => {
+		// Click the Refund button
+		await expect( page ).toClick( 'button.refund-items' );
+
+		// Verify the refund section shows
+		await page.waitForSelector( 'div.wc-order-refund-items', {
+			visible: true,
+		} );
+
+		// Verify Refund via WooCommerce Payments button is displayed
+		await page.waitForSelector( 'button.do-api-refund' );
+
+		// Initiate a refund
+		await expect( page ).toFill( '.refund_order_item_qty', '1' );
+		await expect( page ).toFill( '#refund_reason', 'No longer wanted' );
+
+		await expect( page ).toMatchElement( '.do-api-refund', {
+			text: `Refund ${ orderAmount } via WooCommerce Payments`,
+		} );
+
+		const refundDialog = await expect( page ).toDisplayDialog( async () => {
+			await expect( page ).toClick( 'button.do-api-refund' );
+		} );
+
+		// Accept the refund
+		await refundDialog.accept();
+
+		await uiUnblocked();
+
+		await page.waitForNavigation( { waitUntil: 'networkidle0' } );
+
+		await Promise.all( [
+			// Verify the product line item shows the refunded amount
+			expect( page ).toMatchElement( '.line_cost .refunded', {
+				text: `-${ orderAmount }`,
+			} ),
+
+			// Verify the refund shows in the list with the amount
+			expect( page ).toMatchElement( '.refund > .line_cost', {
+				text: `-${ orderAmount }`,
+			} ),
+
+			// Verify system note was added
+			expect( page ).toMatchElement( '.system-note', {
+				text: `A refund of ${ orderAmount } was successfully processed using WooCommerce Payments. Reason: No longer wanted`,
+			} ),
+		] );
+	} );
+
+	it( 'should be able to view a refunded transaction', async () => {
+		// Pull out and follow the link to avoid working in multiple tabs
+		const paymentDetailsLink = await page.$eval(
+			'p.order_number > a',
+			( anchor ) => anchor.getAttribute( 'href' )
+		);
+		await Promise.all( [
+			page.goto( paymentDetailsLink, {
+				waitUntil: 'networkidle0',
+			} ),
+		] );
+
+		await uiLoaded();
+
+		// Verify the transaction timeline reflects the refund events
+		await Promise.all( [
+			expect( page ).toMatchElement( 'li.woocommerce-timeline-item', {
+				text: `A payment of ${ orderAmount } was successfully refunded.`,
+			} ),
```

Comment:
Good call out--Let me run through it a bit, it could be a sign of flakiness that we may want to address early before we merge this in.

Reply:
I was able to see this failing on each run in non-headless mode (while it passed consistently in headless mode), just to confirm, were you running the tests above in non-headless mode?

Reply:
> Could it be a node version? 

I double checked my Node version and I'm on the same version, `v12.21.0`. I also did a similar process removing `node_modules` and running `npm ci` and the tests still passed.

I am wondering if the error above is because the UI had not fully loaded yet, so I added in another check (since both the header section with payment details and the transaction list load separately). Let me know if this helps with the failures you're seeing.

Reply:
Yay! I am glad to hear that helped! 🎉